src/robusta/utils/cluster_provider_discovery.py

```diff
@@ -18,103 +18,95 @@ class ClusterProviderType(Enum):
     DigitalOcean = "DigitalOcean"
     Unknown = auto()
 
+
 # the value is a regex match of the hostname
 HOSTNAME_MATCH: Dict[ClusterProviderType, str] = {
     ClusterProviderType.Kind: ".*kind.*",
-    ClusterProviderType.RancherDesktop: ".*rancher-desktop.*"
+    ClusterProviderType.RancherDesktop: ".*rancher-desktop.*",
 }
 
 # the value is a node label unique to the provider
 NODE_LABELS: Dict[ClusterProviderType, str] = {
     ClusterProviderType.Minikube: "minikube.k8s.io/name",
     ClusterProviderType.DigitalOcean: "doks.digitalocean.com/version",
     ClusterProviderType.Kops: "kops.k8s.io/instancegroup",
-    ClusterProviderType.Kapsule: "k8s.scaleway.com/kapsule"
+    ClusterProviderType.Kapsule: "k8s.scaleway.com/kapsule",
 }
 
-__CLUSTER_PROVIDER__ = None
-
-
-def get_cluster_provider():
-    global __CLUSTER_PROVIDER__
-    return __CLUSTER_PROVIDER__
-
-
-def discover_cluster_provider():
-    global __CLUSTER_PROVIDER__
-    if not __CLUSTER_PROVIDER__:
-        __CLUSTER_PROVIDER__ = _find_cluster_provider().value
-        logging.info(f"{__CLUSTER_PROVIDER__} cluster detected")
-
-
-def _get_node_label(node, label) -> Optional[str]:
-    try:
-        return node.metadata.labels[label]
-    except KeyError:
-        pass
-    return None
-
 
-def _detect_provider_from_hostname(nodes) -> Optional[ClusterProviderType]:
-    nodes_host_names = [_get_node_label(node, "kubernetes.io/hostname") for node in nodes]
-    for host_name in nodes_host_names:
-        if not host_name:
-            continue
-        for cluster_type in HOSTNAME_MATCH:
-            cluster_hostname_regex = HOSTNAME_MATCH[cluster_type]
-            if len(re.findall(cluster_hostname_regex, host_name)) >= 1:
+class ClusterProviderDiscovery:
```

Comment:
Better avoid of using whole class of @staticmethod in favor of having a normal class with just one instance initiated in module global scope:

```python
   cluster_provider_discovery = ClusterProviderDiscovery()
   __all__ = ["cluster_provider_discovery"]
```